### PR TITLE
Relistan/fix proxied headers

### PR DESCRIPTION
**Pull request by relistan**
The real meat of this is [this commit](https://github.com/Nitro/spacesuit/commit/bd93a82cc8ee9ca4ded1551f29f40f39d666838a). The rest is getting this building on Elixir 1.6 and Erlang/OTP 20.2.2 which is a long overdue update.

The main purpose of this is to fix an issue where headers were being duplicated due to the way Cowboy handles down-casing. This should fix issues with multiples copies of CORS headers being sent. It also fixes a situation where we would overwrite X-Forwarded-For addresses instead of appending.

Additionally, Elixir 1.6 shipped with the official code formatter, so all the code has been reformatted. I you want to review the changes just use the single commit above.

### CODE REVIEW

**Comment by relistan** [comment]
@jopecko awesome! Thanks for digging through that. I want to deploy and test today.

> update elixir version to 1.6.0 in docker/Dockerfile, won't build on master otherwise

Yes, good call.

> after running mix deps.get I have a minor local diff in mix.lock; possibly can be ignored but wanted to mention

Odd, I do not see that. What is the diff?

> seeing a non-fatal warning when running the test suite warning: the dependency :lager_logger requires Elixir ">= 1.1.0 and < 1.5.0" but you are running on v1.6.0

Yes, I saw that, too. I looked at Lager, however, and it seems to be working. Looks like sloppy dependency specification. Thanks!

File: lib/spacesuit.ex [hunk 1]
```diff
@@ -10,23 +10,24 @@ defmodule Spacesuit do
...
-           ]
-         }
-    )
+    {:ok, _} =
+      :cowboy.start_clear(:http, 100, [port: @http_port], %{
+        env: %{
+          dispatch: dispatch
+        },
+        middlewares: [
+          # :cowboy_router, Spacesuit.DebugMiddleware, Spacesuit.AuthMiddleware, :cowboy_handler
```
Comment (epileptic-fish) [inline lib/spacesuit.ex]: Do we need to keep these commented middlewares?

Comment (relistan) [reply lib/spacesuit.ex]: It prevents having to remember where to put the debug middleware when turning it on.

File: lib/spacesuit/proxy_handler.ex [hunk 2]
```diff
@@ -117,26 +116,26 @@ defmodule Spacesuit.ProxyHandler do
...
-    [[_, _, host | _ ] | _ ] = url
+    [[_, _, host | _] | _] = url
 
     (headers || %{})
-      |> Map.put("X-Forwarded-For", peer)
-      |> Map.put("X-Forwarded-Url", url)
-      |> Map.put("X-Forwarded-Host", host)
-      |> Map.drop([ "host", "Host" ])
-      |> Map.to_list
+    |> Map.merge(%{"x-forwarded-for" => peer}, fn _k, a, b -> "#{a}, #{b}" end)
```
Comment (epileptic-fish) [inline lib/spacesuit/proxy_handler.ex]: Does this necessarily maintain the order of the ips?

eg:
```

Would

"X-Forwarded-For: <client>, <proxy1>, <proxy2>" 
+ 
"x-forwarded-For: <other>"
= 
"x-forwarded-for = <client>, <proxy1>, <proxy2>, <other>"

OR

x-forwarded-for = <other>, <client>, <proxy1>, <proxy2>
```
I guess the question is, what governs the ordering here?

Comment (epileptic-fish) [reply lib/spacesuit/proxy_handler.ex]: Apologies for formatting!

Comment (relistan) [reply lib/spacesuit/proxy_handler.ex]: I think it will do the right thing, it seems to maintain order. Thanks!

**Comment by jopecko** [comment]
@relistan nevermind that comment about the dirty `mix.lock`. I think it had something to do with the fact that I didn't run `mix deps.update setup` first. I've since reverted the mod and am not able to reproduce now. Apologies for the false report.

**Pull request merged**